// @pulumi/aws/datazone/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Project = exports.Glossary = exports.getEnvironmentBlueprintOutput = exports.getEnvironmentBlueprint = exports.EnvironmentBlueprintConfiguration = exports.Domain = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Domain = null;
utilities.lazyLoad(exports, ["Domain"], () => require("./domain"));
exports.EnvironmentBlueprintConfiguration = null;
utilities.lazyLoad(exports, ["EnvironmentBlueprintConfiguration"], () => require("./environmentBlueprintConfiguration"));
exports.getEnvironmentBlueprint = null;
exports.getEnvironmentBlueprintOutput = null;
utilities.lazyLoad(exports, ["getEnvironmentBlueprint", "getEnvironmentBlueprintOutput"], () => require("./getEnvironmentBlueprint"));
exports.Glossary = null;
utilities.lazyLoad(exports, ["Glossary"], () => require("./glossary"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:datazone/domain:Domain":
                return new exports.Domain(name, undefined, { urn });
            case "aws:datazone/environmentBlueprintConfiguration:EnvironmentBlueprintConfiguration":
                return new exports.EnvironmentBlueprintConfiguration(name, undefined, { urn });
            case "aws:datazone/glossary:Glossary":
                return new exports.Glossary(name, undefined, { urn });
            case "aws:datazone/project:Project":
                return new exports.Project(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "datazone/domain", _module);
pulumi.runtime.registerResourceModule("aws", "datazone/environmentBlueprintConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "datazone/glossary", _module);
pulumi.runtime.registerResourceModule("aws", "datazone/project", _module);
